### PR TITLE
Feature/myz 8217 file upload component

### PR DESCRIPTION
**Pull request by saldomik**
# [Component] - File Upload - Mobile view without progress bar

----

### Motivation and Context

First step of new file upload component. Mobile view without progress bar

### Priority

- [ ] 1 - Highest
- [ ] 2 - High
- [x] 3 - Medium
- [ ] 4 - Low
- [ ] 5 - Lowest
- [ ] 6 - Not a Priority

### Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Feature (non-breaking change which adds functionality)
- [x] Component (add a Component as approved by Design System)
- [ ] Docs (add documentation)
- [ ] Chore (changes that adds small enhancement)
- [ ] Breaking (fix or feature that would cause existing functionality to not work as expected)

### Abstract

https://app.abstract.com/projects/fd370780-e659-11e8-99dc-0d08537c5fde/branches/17e87d7b-3982-433c-9954-0a9e19c7f55b/commits/latest/files/9f586aa0-bd5b-4c12-9541-aed2e98a7a9c/layers/2CCDB253-4F0E-4B98-B612-7400A244CA5B?collectionId=ffbeeba6-ceda-4fa2-9924-634e795aa64e&collectionLayerId=5a1bbddb-4ce6-4950-b3cb-8a0c40db4a60

### Note

----

## Component and Fix approval flow to Master branch

A Pull Request to be merged must have two approvals, one from a member of the product team who developed it and another from a member of the dst dev team other than the team representative.

### CODE REVIEW

File: src/components/z-file-upload/index.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+@Component({
+  tag: "z-file-upload",
+  styleUrl: "styles.css",
+  shadow: true,
+})
+export class ZFileUpload {
+  private input: HTMLInputElement;
+
+  @Prop() variant: ButtonVariantEnum;
```
Comment (vpasquino) [inline src/components/z-file-upload/index.tsx:20]: JSDoc

File: src/components/z-file-upload/index.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+  @State() files: number = 0;
+
+  @Listen("removeFile")
+  removeFileListener() {
+    this.files--;
+  }
+
+  @Event() fileInput: EventEmitter;
```
Comment (vpasquino) [inline src/components/z-file-upload/index.tsx:29]: JSDoc

File: src/components/z-file-upload/index.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  }
+
+  renderInput() {
+    return [
+      <input
+        onChange={() => this.fileInputHandler()}
+        type="file"
+        id="fileElem"
+        multiple
+        accept="image/*"
```
Comment (vpasquino) [inline src/components/z-file-upload/index.tsx:81]: Perché solo immagini?

File: src/components/z-file/index.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+@Component({
+  tag: 'z-file',
+  styleUrl: 'styles.css',
+  shadow: true
+})
+
+export class ZFile {
+  @Element() el: HTMLElement;
+
+  @Event() removeFile: EventEmitter
```
Comment (vpasquino) [inline src/components/z-file/index.tsx:12]: JSDoc

File: src/components/z-file-upload/index.tsx (hunk 3 above)
Comment (saldomik) [reply src/components/z-file-upload/index.tsx:81]: perchè ho sbagliato!

File: src/components/z-file-upload/index.tsx [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+  @Listen("removeFile")
+  removeFileListener() {
+    this.files--;
+  }
+
+  @Event() fileInput: EventEmitter;
+  fileInputHandler() {
+    if (this.input.files.length) {
+      this.fileInput.emit(this.input.files);
+      this.files++;
```
Comment (vpasquino) [inline src/components/z-file-upload/index.tsx:33]: Non `this.files++;` ma `this.files += this.input.files.length;`

**Review by vpasquino — APPROVED** [review]

**Review by AliGamb — APPROVED** [review]

**Pull request merged**